src/optimizer/rules/implementation_rules.cpp

```diff
@@ -523,6 +523,60 @@ void LogicalInnerJoinToPhysicalInnerHashJoin::Transform(common::ManagedPointer<O
   }
 }
 
+///////////////////////////////////////////////////////////////////////////////
+/// LogicalSemiJoinToPhysicalSemiLeftHashJoin
+///////////////////////////////////////////////////////////////////////////////
+LogicalSemiJoinToPhysicalSemiLeftHashJoin::LogicalSemiJoinToPhysicalSemiLeftHashJoin() {
+  type_ = RuleType::SEMI_JOIN_TO_HASH_JOIN;
+
+  // Make three node types for pattern matching
+  auto left_child(new Pattern(OpType::LEAF));
+  auto right_child(new Pattern(OpType::LEAF));
+
+  // Initialize a pattern for optimizer to match
+  match_pattern_ = new Pattern(OpType::LOGICALSEMIJOIN);
+
+  // Add node - we match join relation R and S as well as the predicate exp
+  match_pattern_->AddChild(left_child);
+  match_pattern_->AddChild(right_child);
+}
+
+bool LogicalSemiJoinToPhysicalSemiLeftHashJoin::Check(common::ManagedPointer<OperatorNode> plan,
+                                                    OptimizationContext *context) const {
+  (void)context;
+  (void)plan;
+  return true;
+}
+
+void LogicalSemiJoinToPhysicalSemiLeftHashJoin::Transform(common::ManagedPointer<OperatorNode> input,
+                                                        std::vector<std::unique_ptr<OperatorNode>> *transformed,
+                                                        UNUSED_ATTRIBUTE OptimizationContext *context) const {
+  // first build an expression representing hash join
+  const auto semi_join = input->GetOp().As<LogicalSemiJoin>();
+
+  auto children = input->GetChildren();
+  TERRIER_ASSERT(children.size() == 2, "Left Semi Join should have two child");
+  auto left_group_id = children[0]->GetOp().As<LeafOperator>()->GetOriginGroup();
+  auto right_group_id = children[1]->GetOp().As<LeafOperator>()->GetOriginGroup();
+  auto &left_group_alias = context->GetOptimizerContext()->GetMemo().GetGroupByID(left_group_id)->GetTableAliases();
+  auto &right_group_alias = context->GetOptimizerContext()->GetMemo().GetGroupByID(right_group_id)->GetTableAliases();
+  std::vector<common::ManagedPointer<parser::AbstractExpression>> left_keys;
+  std::vector<common::ManagedPointer<parser::AbstractExpression>> right_keys;
+
+  std::vector<AnnotatedExpression> join_preds = semi_join->GetJoinPredicates();
+  OptimizerUtil::ExtractEquiJoinKeys(join_preds, &left_keys, &right_keys, left_group_alias, right_group_alias);
+
+  TERRIER_ASSERT(right_keys.size() == left_keys.size(), "# left/right keys should equal");
+  std::vector<std::unique_ptr<OperatorNode>> child;
+  child.emplace_back(children[0]->Copy());
```

Comment:
Is this right? Why are we not using the builder and std::move ?

Reply:
This is correct, to be consistent with LogicalInnerJoinToPhysicalInnerHashJoin

Reply:
I think it is wrong there too. Why are we copying the child nodes when we are not using them anywhere later?
Just call move on them after popping them from the vector. 
It being wrong in one other place doesn't make it right.